More THM doco: last PPs and some components

GiudGiud opened the pull request:
refs https://github.com/idaholab/moose/issues/19737

GiudGiud committed:
Add missing documentation to the last few THM components missing docs
refs #19737

GiudGiud committed:
Add missing documentation for some THM's postprocessors
refs #19737

Address review comments

moosebuild commented:
Job [Coverage](https://civet.inl.gov/job/1662037/) on fac42e3 wanted to post the following:


## Framework coverage

<table>
  <thead>
    <tr>
      <th></th>
      <th>d2ef20</th>
      <th colspan=4>#25023 fac42e</th>
    </tr>
    <tr>
      <th></th>
      <th>Total</th>
      <th>Total</th>
      <th>+/-</th>
      <th>New</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><b>Rate</b></td>
      <td>85.59%</td>
      <td>85.59%</td>
      <td>+0.00%</td>
      <td>-</td>
    </tr>
    <tr>
      <td><b>Hits</b></td>
      <td>89529</td>
      <td>89530</td>
      <td>+1</td>
      <td>0</td>
    </tr>
    <tr>
      <td><b>Misses</b></td>
      <td>15073</td>
      <td>15072</td>
      <td>-1</td>
      <td>0</td>
    </tr>
  </tbody>
</table>

[Diff coverage report](https://mooseframework.inl.gov/docs/PRs/25023/coverage_diff/framework)

[Full coverage report](https://mooseframework.inl.gov/docs/PRs/25023/coverage/framework)


## Modules coverage

### Thermal hydraulics

<table>
  <thead>
    <tr>
      <th></th>
      <th>d2ef20</th>
      <th colspan=4>#25023 fac42e</th>
    </tr>
    <tr>
      <th></th>
      <th>Total</th>
      <th>Total</th>
      <th>+/-</th>
      <th>New</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><b>Rate</b></td>
      <td>90.60%</td>
      <td>90.60%</td>
      <td>+0.00%</td>
      <td>100.00%</td>
    </tr>
    <tr>
      <td><b>Hits</b></td>
      <td>13017</td>
      <td>13019</td>
      <td>+2</td>
      <td>2</td>
    </tr>
    <tr>
      <td><b>Misses</b></td>
      <td>1351</td>
      <td>1351</td>
      <td>-</td>
      <td>0</td>
    </tr>
  </tbody>
</table>

[Diff coverage report](https://mooseframework.inl.gov/docs/PRs/25023/coverage_diff/thermal_hydraulics)

[Full coverage report](https://mooseframework.inl.gov/docs/PRs/25023/coverage/thermal_hydraulics)

## Full coverage reports

<details>
<summary>Reports</summary>

- [`framework`](https://mooseframework.inl.gov/docs/PRs/25023/coverage/framework)
- [`chemical_reactions`](https://mooseframework.inl.gov/docs/PRs/25023/coverage/chemical_reactions)
- [`combined`](https://mooseframework.inl.gov/docs/PRs/25023/coverage/combined)
- [`contact`](https://mooseframework.inl.gov/docs/PRs/25023/coverage/contact)
- [`electromagnetics`](https://mooseframework.inl.gov/docs/PRs/25023/coverage/electromagnetics)
- [`external_petsc_solver`](https://mooseframework.inl.gov/docs/PRs/25023/coverage/external_petsc_solver)
- [`fluid_properties`](https://mooseframework.inl.gov/docs/PRs/25023/coverage/fluid_properties)
- [`fsi`](https://mooseframework.inl.gov/docs/PRs/25023/coverage/fsi)
- [`functional_expansion_tools`](https://mooseframework.inl.gov/docs/PRs/25023/coverage/functional_expansion_tools)
- [`geochemistry`](https://mooseframework.inl.gov/docs/PRs/25023/coverage/geochemistry)
- [`heat_conduction`](https://mooseframework.inl.gov/docs/PRs/25023/coverage/heat_conduction)
- [`level_set`](https://mooseframework.inl.gov/docs/PRs/25023/coverage/level_set)
- [`misc`](https://mooseframework.inl.gov/docs/PRs/25023/coverage/misc)
- [`navier_stokes`](https://mooseframework.inl.gov/docs/PRs/25023/coverage/navier_stokes)
- [`optimization`](https://mooseframework.inl.gov/docs/PRs/25023/coverage/optimization)
- [`peridynamics`](https://mooseframework.inl.gov/docs/PRs/25023/coverage/peridynamics)
- [`phase_field`](https://mooseframework.inl.gov/docs/PRs/25023/coverage/phase_field)
- [`porous_flow`](https://mooseframework.inl.gov/docs/PRs/25023/coverage/porous_flow)
- [`ray_tracing`](https://mooseframework.inl.gov/docs/PRs/25023/coverage/ray_tracing)
- [`rdg`](https://mooseframework.inl.gov/docs/PRs/25023/coverage/rdg)
- [`reactor`](https://mooseframework.inl.gov/docs/PRs/25023/coverage/reactor)
- [`richards`](https://mooseframework.inl.gov/docs/PRs/25023/coverage/richards)
- [`scalar_transport`](https://mooseframework.inl.gov/docs/PRs/25023/coverage/scalar_transport)
- [`solid_properties`](https://mooseframework.inl.gov/docs/PRs/25023/coverage/solid_properties)
- [`stochastic_tools`](https://mooseframework.inl.gov/docs/PRs/25023/coverage/stochastic_tools)
- [`tensor_mechanics`](https://mooseframework.inl.gov/docs/PRs/25023/coverage/tensor_mechanics)
- [`thermal_hydraulics`](https://mooseframework.inl.gov/docs/PRs/25023/coverage/thermal_hydraulics)
- [`xfem`](https://mooseframework.inl.gov/docs/PRs/25023/coverage/xfem)

</details>



This comment will be updated on new commits.

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/1661993/) on fac42e3 wanted to post the following:

View the site [here](https://mooseframework.inl.gov/docs/PRs/25023/site)

This comment will be updated on new commits.

GiudGiud committed:
Make a clearer error message for general field transfers

GiudGiud committed:
Add docs for heat conduction model used in THM HS boundary components

GiudGiud commented on modules/thermal_hydraulics/doc/content/source/postprocessors/ADHeatStructureEnergy3D.md:
```diff
@@ -1,20 +1,17 @@
 
-!! Replace these lines with information regarding the ADHeatStructureEnergy3D object.
+!equation
+E = \int_\Omega \rho cp (T - T_{ref}) d\Omega
```
```suggestion
E = \int_\Omega \rho c_p (T - T_{ref}) d\Omega
```

GiudGiud commented on modules/thermal_hydraulics/doc/content/source/postprocessors/ADHeatStructureEnergy.md:
```diff
@@ -1,20 +1,18 @@
 
-!! Replace these lines with information regarding the ADHeatStructureEnergy object.
+!equation
+E = \int_\Omega \rho cp (T - T_{ref}) n_{ }units \text{plate depth} d\Omega
```
```suggestion
E = \int_\Omega \rho c_p (T - T_{ref}) n_{ }units \text{plate depth} d\Omega
```

GiudGiud commented on modules/thermal_hydraulics/doc/content/source/postprocessors/ADHeatStructureEnergyRZ.md:
```diff
@@ -1,20 +1,22 @@
+The energy stored in these heat structures is calculated as:
+
+!equation
+E = \int_\Omega \rho cp (T - T_{ref}) n_{ }units r d\Omega
```
```suggestion
E = \int_\Omega \rho c_p (T - T_{ref}) n_{ }units r d\Omega
```

GiudGiud commented on modules/thermal_hydraulics/doc/content/source/postprocessors/ADHeatStructureEnergy.md:
```diff
@@ -1,20 +1,18 @@
+E = \int_\Omega \rho cp (T - T_{ref}) n_{ }units \text{plate depth} d\Omega
 
-## Example Input File Syntax
+where $E$ is the stored energy, $\rho$ the medium density, $cp$ the medium heat capacity,
```
```suggestion
where $E$ is the stored energy, $\rho$ the medium density, $c_p$ the medium specific heat capacity,
```

GiudGiud commented on modules/thermal_hydraulics/doc/content/source/postprocessors/ADHeatStructureEnergy3D.md:
```diff
@@ -1,20 +1,17 @@
+E = \int_\Omega \rho cp (T - T_{ref}) d\Omega
 
-## Example Input File Syntax
+where $E$ is the stored energy, $\rho$ the medium density, $cp$ the medium heat capacity,
```
```suggestion
where $E$ is the stored energy, $\rho$ the medium density, $c_p$ the medium specific heat capacity,
```

GiudGiud commented on modules/thermal_hydraulics/doc/content/source/postprocessors/ADHeatStructureEnergyRZ.md:
```diff
@@ -1,20 +1,22 @@
+E = \int_\Omega \rho cp (T - T_{ref}) n_{ }units r d\Omega
 
-!! Replace these lines with information regarding the ADHeatStructureEnergyRZ object.
+where $E$ is the stored energy, $\rho$ the medium density, $cp$ the medium heat capacity,
```
```suggestion
where $E$ is the stored energy, $\rho$ the medium density, $c_p$ the medium specific heat capacity,
```

joshuahansel commented on modules/thermal_hydraulics/doc/content/source/postprocessors/ADHeatStructureEnergy.md:
```diff
@@ -1,20 +1,18 @@
 
-!! Replace these lines with information regarding the ADHeatStructureEnergy object.
+!equation
+E = \int_\Omega \rho c_p (T - T_{ref}) n_{units} \text{plate depth} d\Omega
```
```suggestion
E = \int_\Omega \rho c_p (T - T_{ref}) n_{units} d d\Omega
```

joshuahansel commented on modules/thermal_hydraulics/doc/content/source/postprocessors/ADHeatStructureEnergy.md:
```diff
@@ -1,20 +1,18 @@
-## Example Input File Syntax
+where $E$ is the stored energy, $\rho$ the medium density, $c_p$ the medium specific heat capacity,
+$T$ the medium temperature, $T_{ref}$ the reference temperature (reference for the enthalpy)
+and $n_{units}$ the number of units this heat structure represents and the plate depth is the
+size of uni-dimensional heat structures modeled.
```
```suggestion
and $n_{units}$ the number of units this heat structure represents, and $d$ is the plate depth, i.e., the
size of uni-dimensional heat structures modeled.
```

joshuahansel commented on modules/thermal_hydraulics/doc/content/source/postprocessors/ADHeatStructureEnergy.md:
```diff
@@ -1,20 +1,18 @@
+size of uni-dimensional heat structures modeled.
 
-!! Describe and include an example of how to use the ADHeatStructureEnergy object.
+This volumetric integral is computed numerically using the system's quadrature.
```
Question because I don't know: are quadrature rules associated with systems? I remember something about block-dependent rules.

GiudGiud replied:
so yeah no, not a system thing

joshuahansel commented on modules/thermal_hydraulics/doc/content/source/postprocessors/ADHeatStructureEnergyRZ.md:
```diff
@@ -1,20 +1,22 @@
+!alert warning
+With the recent development of general RZ coordinates, this object along with all THM's "RZ"-specific
+objects will soon be deprecated in favor of more general 2D objects used in general RZ coordinates.
+Stay tuned!
```
I should be having time to get back to this soon! 🤞

GiudGiud replied:
between that and merging AD and non_AD using templates, THM could be half the code lolol

joshuahansel commented on modules/thermal_hydraulics/doc/content/source/postprocessors/NodalEnergyFluxPostprocessor.md:
```diff
@@ -1,20 +1,14 @@
 
-!! Describe and include an example of how to use the NodalEnergyFluxPostprocessor object.
+where $\alpha$ is the phase fraction, $\rho$ is the phase density, $u$ is the phase velocity,
+$A$ the local area and $H$ the specific total enthalpy.
```
```suggestion
$A$ the local area, and $H$ the specific total enthalpy.
```

joshuahansel replied:
What's your position on the Oxford comma? No one seems to use it, but everyone seems to think it's the right thing to do. I prefer to be a hero.

GiudGiud replied:
my English teachers never got on to me for this so I m fairly neutral

roystgnr replied:
I use it!  If you ever see me write that famous "We invited the strippers, JFK and Stalin" sentence, don't assume I forgot a comma; rush to a news website right away and see what dirt has just been declassified.

GiudGiud commented on modules/thermal_hydraulics/doc/content/source/postprocessors/ADHeatStructureEnergy.md:
```diff
@@ -1,20 +1,18 @@
+size of uni-dimensional heat structures modeled.
 
-!! Describe and include an example of how to use the ADHeatStructureEnergy object.
+This volumetric integral is computed numerically using the system's quadrature.
```
```suggestion
This volumetric integral is computed numerically using the [local quadrature](syntax/Executioner/Quadrature/index.md)
```

GiudGiud committed:
Address Josh's review

- more accurate comment about using the local quadrature
- formatting fixes
- oxford comma

Co-authored-by: joshuahansel <joshua.hansel@inl.gov>

Apply suggestions from code review

Co-authored-by: joshuahansel <joshua.hansel@inl.gov>

GiudGiud commented:
Test failure is fake

joshuahansel commented on modules/thermal_hydraulics/doc/content/source/postprocessors/ADHeatStructureEnergy3D.md:
```diff
@@ -1,20 +1,17 @@
+Only a single unit of the heat structure is considered.
 
-!! Describe and include an example of how to use the ADHeatStructureEnergy3D object.
+This volumetric integral is computed numerically using the system's quadrature.
```
```suggestion
This volumetric integral is computed numerically using the [local quadrature](syntax/Executioner/Quadrature/index.md).
```

joshuahansel commented on modules/thermal_hydraulics/doc/content/source/postprocessors/ADHeatStructureEnergyRZ.md:
```diff
@@ -1,20 +1,22 @@
+and $n_{units}$ the number of units this heat structure represents and $r$ the local circumference.
 
-## Example Input File Syntax
+This volumetric integral is computed numerically using the system's quadrature.
```
```suggestion
This volumetric integral is computed numerically using the [local quadrature](syntax/Executioner/Quadrature/index.md).
```

GiudGiud commented:
done
